File: package/node/src/modules/token.ts
import type { Transport } from "../core/transport";
import type { SessionManager } from "../core/session";
import type { TokenResponse } from "../types";
import { SessionExpiredError } from "../errors";

interface TokenModuleDeps {
  transport: Transport;
  session: SessionManager;
}

class TokenModule {
  private transport: Transport;
  private session: SessionManager;

  constructor(deps: TokenModuleDeps) {
    this.transport = deps.transport;
    this.session = deps.session;
  }

  async refresh(): Promise<void> {
    const refreshToken = this.session.getRefreshToken();

    if (!refreshToken) {
      throw new SessionExpiredError();
    }

    const response = await this.transport.post<TokenResponse>(
      "/api/v1/auth/refresh",
      { refreshToken },
    );

    this.session.setAccessToken(response.accessToken, response.expiresIn);
  }

  async revoke(): Promise<void> {
    const accessToken = this.session.getAccessToken();

    if (!accessToken) {
      return;
    }

    try {
      await this.transport.post(
        "/api/v1/auth/token/revoke",
        undefined,
        accessToken,
      );
    } catch {}

    this.session.clear();
  }
}

export { TokenModule };

File: package/python/aether_identity/modules/token.py
"""Aether Identity Python SDK - Token module."""

from ..core.transport import Transport
from ..core.session import SessionManager
from ..types import TokenResponse


class TokenModule:
    """Token operations module."""

    def __init__(self, transport: Transport, session_manager: SessionManager):
        """Initialize token module.

        Args:
            transport: HTTP transport instance
            session_manager: Session manager instance
        """
        self._transport = transport
        self._session = session_manager

    def refresh(self) -> TokenResponse:
        """Refresh the access token.

        Returns:
            New token response
        """
        refresh_token = self._session.get_refresh_token()

        if not refresh_token:
            raise ValueError("No refresh token available")

        data = {"refreshToken": refresh_token}

        response = self._transport.post("/api/v1/auth/refresh", data)

        tokens = TokenResponse.model_validate(response)
        self._session.set_tokens(tokens)

        return tokens

    def revoke(self) -> None:
        """Revoke the current token.

        Clears session regardless of API success.
        """
        token = self._session.get_access_token()

        if token:
            try:
                self._transport.post("/api/v1/auth/token/revoke", access_token=token)
            except Exception:
                pass

        self._session.clear()
